engine/src/main/java/org/hibernate/search/engine/search/query/dsl/SearchQueryOptionsStep.java

```diff
@@ -35,11 +41,12 @@
  * @param <AF> The type of factory used to create aggregations in {@link #aggregation(AggregationKey, Function)}.
  */
 public interface SearchQueryOptionsStep<
-				S extends SearchQueryOptionsStep<?, H, LOS, SF, AF>,
+				S extends SearchQueryOptionsStep<?, H, LOS, SF, AF, HGLF>,
 				H,
 				LOS,
 				SF extends SearchSortFactory,
-				AF extends SearchAggregationFactory
+				AF extends SearchAggregationFactory,
+				HGLF extends QueryHighlightBuilder
```

Comment:
So the interface name needs some love, as it's not very consistent with the ones above. Maybe `SearchHighlighterFactory`? That would need to be an actual factory, though, not a builder.

That being said... adding generics like that is a backwards-incompatible change :/ We probably have a mention of generics being liable to change without warning somewhere in the javadoc, but nothing visible enough to make me comfortable with such a change.

Let's try without the additional generics if we can?